chain/indexer/integrated/manager.go

```diff
@@ -104,20 +110,17 @@ func (i *Manager) TipSet(ctx context.Context, ts *types.TipSet, options ...index
 		return true, nil
 	}
 
-	var modelResults []*indexer.ModelResult
-	success := true
-	// collect all the results, recording if any of the tasks were skipped or errored
-	for res := range taskResults {
-		select {
-		case fatal := <-taskErrors:
-			lg.Errorw("fatal indexer error", "error", fatal)
-			return false, fatal
-		default:
+	success := atomic.NewBool(true)
+	grp, ctx := errgroup.WithContext(ctx)
```

Comment:
introduction of errgroup fixes deadlock by allowing both channels to be read from simultaneously. Prior to this `taskErrors` was only read when a `taskResults` was received. If there wasn't a `taskResult` then the tipset indexer deadlocked writing to the `taskErrors` channel since nothing was reading from it.